### PR TITLE
add info.json for dactyl_promicro

### PR DESCRIPTION
**Pull request by immeraufdemhund**
Add info.json for dactyl_promicro

## Description

When using the online firmware builder info.json is used to display keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by immeraufdemhund** [comment]
FWIW how I tested is that I use the info.json for the manuforms and the dactyl in a WPF app that I wrote. I noticed how the manuforms were written then looked at the dactyl as a guide. This is my programs interpretation of the manuform 6x6
image

and the dactyl_promicro (what I submitted the PR for... I don't have height working in my program)
image

**Comment by yanfali** [comment]
@immeraufdemhund https://config.qmk.fm then hold the following keys in sequence: ctrl + shift + i
you will be prompted for an info.json file which you can then test in the uI.

**Comment by immeraufdemhund** [comment]
**GASP** what a beautiful thing!
image

**Comment by noroadsleft** [comment]
My best guess of the layout is:

image

Based on [this last commit from the submitting PR](https://github.com/qmk/qmk_firmware/pull/5113/commits/1828a7e7fe3db2d0bbdccdd115f38e3d2ee60451#diff-f308d3c6041909837d760ebe709beaccR16-R23).

Additionally, to my mind the `keymap.c` should be formatted as:

```c
    [_QWERTY] = LAYOUT_6x6(
        KC_ESC ,  KC_1  , KC_2   ,  KC_3  ,  KC_4  ,  KC_5  ,           KC_6  ,  KC_7  ,  KC_8  ,  KC_9  ,  KC_0  , KC_BSPC,
        KC_TAB ,  KC_Q  , KC_W   ,  KC_E  ,  KC_R  ,  KC_T  ,           KC_Y  ,  KC_U  ,  KC_I  ,  KC_O  ,  KC_P  , KC_MINS,
        KC_LSFT,  KC_A  , KC_S   ,  KC_D  ,  KC_F  ,  KC_G  ,           KC_H  ,  KC_J  ,  KC_K  ,  KC_L  , KC_SCLN, KC_QUOT,
        KC_LCTL,  KC_Z  , KC_X   ,  KC_C  ,  KC_V  ,  KC_B  ,           KC_N  ,  KC_M  , KC_COMM, KC_DOT , KC_SLSH, KC_BSLASH,
        KC_PLUS, KC_LEFT, KC_UP  , KC_DOWN, KC_RGHT,  RAISE ,           LOWER , KC_HOME, KC_PGUP, KC_PGDN, KC_END , KC_EQL,
                                   KC_SPC , _______, KC_LALT,          KC_PSCR, _______, KC_ENT,
                                                     KC_PSCR,          KC_AT  ,
                                                     KC_LGUI,          KC_GRAVE
    ),
```

... though I consider that out-of-scope for this PR.

**Comment by immeraufdemhund** [comment]
I disagree slightly with the layout. I build these keyboards for OhKeycaps and your render is a little difficult to follow. Here is a random picture from the internet of said dactyl_manuform
image

that said the x,y of the layout in my PR I feel is more accurate (however it is very possible that the keys are out of order)

**Comment by immeraufdemhund** [comment]
I just looked at my layout.c and I see why you feel the x/y of yours. I should have taken the time to put that in as well in my PR. I don't mind to reformat it.  I basically need to go to the dactyl keyboard and put them on the same plane:

image

but that layout is more accurate

**Comment by immeraufdemhund** [comment]
i'll ammend my commit and fix that too, cause truthfully it bugs me

**Comment by immeraufdemhund** [comment]
rebased on master and added another commit to format a few things

**Comment by immeraufdemhund** [comment]
*sigh*... failed build

File: keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c [hunk 1]
```diff
@@ -12,36 +9,44 @@ extern keymap_config_t keymap_config;
...
+    ),//KC_END ,KC_EQL ,KC_LALT
+    [_LOWER] = LAYOUT_dactyl(
+        _______,KC_F1  ,KC_F2  ,KC_F3  ,KC_F4  ,KC_F5  ,                                        KC_F6  ,KC_F7  ,KC_F8  ,KC_F9  ,KC_F10 ,KC_F11 ,
+        _______,KC_EXLM,KC_AT  ,KC_LBRC,KC_RBRC,KC_PIPE,                                        KC_UP  ,KC_P7  ,KC_P8  ,KC_P9  ,KC_PAST,KC_F12 ,
+        _______,KC_HASH,KC_DLR ,KC_LPRN,KC_RPRN,KC_GRV ,                                        KC_DOWN,KC_P4  ,KC_P5  ,KC_P6  ,KC_PPLS,_______,
+        _______,KC_PERC,KC_CIRC,KC_LBRC,KC_RBRC,KC_TILD,                                        KC_AMPR,KC_P1  ,KC_P2  ,KC_P3  ,KC_PSLS,_______,
+        RESET  ,_______,_______,_______,_______,                                                        _______,KC_PDOT,KC_P0  ,KC_PEQL,_______,
+                                                        _______,_______,       _______,_______,
+                                                                _______,       _______,
+                                                _______,_______,_______,       _______,_______,_______,
```
Comment (kuchosauronad0) [inline keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]: Trailing , at the end of the line here. Should be 
```
                                                _______,_______,_______,       _______,_______,_______
```

File: keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c [hunk 2]
```diff
@@ -12,36 +9,44 @@ extern keymap_config_t keymap_config;
...
+    [_RAISE] = LAYOUT_dactyl(
+        _______,_______,_______,_______,_______,_______,                                        _______,_______,_______,_______,_______,_______,
+        _______,_______,_______,KC_MS_U,_______,_______,                                        _______,_______,_______,_______,_______,_______,
+        _______,_______,KC_MS_L,KC_MS_D,KC_MS_R,_______,                                        _______,_______,_______,_______,_______,KC_MPLY,
+        _______,_______,_______,_______,_______,_______,                                        _______,_______,KC_MPRV,KC_MNXT,_______,_______,
+        RESET  ,_______,_______,KC_BTN1,KC_BTN2,                                                        KC_VOLU,KC_VOLD,KC_MUTE,_______,_______,
+                                                        _______,_______,       _______,_______,
+                                                                _______,       _______,
+                                                _______,_______,_______,       _______,_______,KC_WBAK,
+    ),
```
Comment (kuchosauronad0) [inline keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]:
```suggestion
    )
```

File: keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c [hunk 3]
```diff
@@ -12,36 +9,44 @@ extern keymap_config_t keymap_config;
...
+    [_RAISE] = LAYOUT_dactyl(
+        _______,_______,_______,_______,_______,_______,                                        _______,_______,_______,_______,_______,_______,
+        _______,_______,_______,KC_MS_U,_______,_______,                                        _______,_______,_______,_______,_______,_______,
+        _______,_______,KC_MS_L,KC_MS_D,KC_MS_R,_______,                                        _______,_______,_______,_______,_______,KC_MPLY,
+        _______,_______,_______,_______,_______,_______,                                        _______,_______,KC_MPRV,KC_MNXT,_______,_______,
+        RESET  ,_______,_______,KC_BTN1,KC_BTN2,                                                        KC_VOLU,KC_VOLD,KC_MUTE,_______,_______,
+                                                        _______,_______,       _______,_______,
+                                                                _______,       _______,
+                                                _______,_______,_______,       _______,_______,KC_WBAK,
```
Comment (kuchosauronad0) [inline keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]:
```suggestion
                                                _______,_______,_______,       _______,_______,KC_WBAK
```

File: keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c [hunk 4]
```diff
@@ -12,36 +9,44 @@ extern keymap_config_t keymap_config;
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_QWERTY] = LAYOUT_dactyl(
+        KC_EQL , KC_1  ,KC_2   , KC_3  , KC_4  , KC_5  ,                                        KC_6   , KC_7  , KC_8  , KC_9  , KC_0  ,KC_BSPC,
+        KC_DEL , KC_Q  ,KC_W   , KC_E  , KC_R  , KC_T  ,                                        KC_Y   , KC_U  , KC_I  , KC_O  , KC_P  ,KC_MINS,
+        KC_BSPC, KC_A  ,KC_S   , KC_D  , KC_F  , KC_G  ,                                        KC_H   , KC_J  , KC_K  , KC_L  ,LT2_COL,LG_QUOT,
+        KC_LSFT, CTL_Z ,KC_X   , KC_C  , KC_V  , KC_B  ,                                        KC_N   , KC_M  ,KC_COMM, KC_DOT,CTL_SLS,KC_RSFT,
+        ONE_GRV,KC_QUOT,ALT_SHFT,KC_LEFT,KC_RGHT,                                                       KC_UP  ,KC_DOWN,KC_LBRC,KC_RBRC,TT(_LOWER),
+                                                        ALT_MENU,KC_LGUI,       KC_RALT,CTL_ESC,
+                                                                KC_HOME,        KC_PGUP,
+                                                KC_SPC ,KC_BSPC,KC_END ,        KC_PGDN,KC_TAB ,KC_ENT ,
```
Comment (kuchosauronad0) [inline keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]:
```suggestion
                                                KC_SPC ,KC_BSPC,KC_END ,        KC_PGDN,KC_TAB ,KC_ENT
```

**Comment by immeraufdemhund** [comment]
I should have built it and tested it earlier on my dactyl keyboard. the matrix is self is wrong. I'll need a little bit of time to diagnose and properly get the matrix down before I fix the requested items. Thank you for pointing them out. Since this is something I use often it wont take long for me to get to

File: keyboards/handwired/dactyl_promicro/dactyl_promicro.h [hunk 5]
```diff
@@ -11,29 +11,28 @@
...
+        {  L30, L31, L32, L33, L34, L35  },                             \
+        {  L40, L41, L42, L43, L44, L45  },                             \
+        {KC_NO, L51, L52, L53, L54, L55  },                             \
+                                                                        \
+        { R00, R01, R02, R03, R04, R05   },                             \
+        { R10, R11, R12, R13, R14, R15   },                             \
+        { R20, R21, R22, R23, R24, R25   },                             \
+        { R30, R31, R32, R33, R34, R35   },                             \
+        { R40, R41, R42, R43, R44, R45   },                             \
+        { R50, R51, R52, R53, R54, KC_NO },                             \
```
Comment (kuchosauronad0) [inline keyboards/handwired/dactyl_promicro/dactyl_promicro.h]:
```suggestion
       { R50, R51, R52, R53, R54, KC_NO }                             \
```

**Comment by zvecr** [comment]
As this change has updated a LAYOUT macro to a new name and different "shape" it breaks backwards compatibility. As there is no way to provide compatibility for existing users via an alisas, I have added the "Breaking Change" label. One possible way round this is to maintain the existing 6x6 macro.

**Comment by immeraufdemhund** [comment]
svecr I will change it back. that's not a problem

**Comment by immeraufdemhund** [comment]
changed things back, compiled it and flashed to my dactyl_promicro, so far it is working well. This is ready IMO

**Comment by immeraufdemhund** [comment]
I'm on my phone right now, going on a trip. I am fairly certain I used the original and my builds with the same wiring and they worked the same. I can verify when I get home over the weekend.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by immeraufdemhund** [comment]
How do you override master_right and ee_hands? I can't find that option in the web

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: keyboards/handwired/dactyl_promicro/dactyl_promicro.h [hunk 6]
```diff
@@ -11,29 +11,28 @@
 #endif
 #endif
 
-
 #define LAYOUT_6x6(\
```
[the remark this answers is not in the document]
Comment (immeraufdemhund) [reply keyboards/handwired/dactyl_promicro/dactyl_promicro.h]: @zvecr  how do you mean not backwards compatible? like with the wiring? or just the define is different

**Comment by ideasman42** [comment]
Any updates on this? I just purchased one of these keyboards and ended up needing to make my own layout define since I wasn't aware of this PR:

https://github.com/ideasman42/qmk_firmware/commit/6d309967bd3c3aa395eda0955f5cfa61d71a48bf#diff-72ee90ef7918acc1ee4a1ba1720f2f0b

**Comment by immeraufdemhund** [comment]
here is picture of what i'm trying to move around which better matches the actual layout in text form (which is also what `info.json` is based from) 
image

**Comment by zvecr** [comment]
@immeraufdemhund that image perfectly depicts how the macro is not backwards compatible, the order of keys has changed, which will effect existing user keymaps. Configurator cannot change EE_HANDS at the moment, but again, as its a behaviour change, existing users will not be informed that they now have to flash extra files to get their keyboards to work.

**Comment by immeraufdemhund** [comment]
But the matrix position is the same. They would compile and problem solved right? I don't understand this backwards compatible apparently

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by zvecr** [comment]
Users dont express keymaps in matrix positions, they use the LAYOUT macros, thus if they put their space in what was L45, it would now be mapped to R41 which is on the other half of the keyboard.

File: keyboards/handwired/dactyl_promicro/dactyl_promicro.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (immeraufdemhund) [reply keyboards/handwired/dactyl_promicro/dactyl_promicro.h]: reverted my rearranging of keys as the info.json is correct and just formatted default

File: keyboards/handwired/dactyl_promicro/keymaps/default/config.h [hunk 7]
```diff
@@ -18,8 +18,4 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 #define USE_SERIAL
-
```
[the remark this answers is not in the document]
Comment (immeraufdemhund) [reply keyboards/handwired/dactyl_promicro/keymaps/default/config.h]: reverted this file all together

File: keyboards/handwired/dactyl_promicro/dactyl_promicro.h (hunk 5 above)
Comment (immeraufdemhund) [reply keyboards/handwired/dactyl_promicro/dactyl_promicro.h]: fixed

File: keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c (hunk 4 above)
Comment (immeraufdemhund) [reply keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]: fixed

File: keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c (hunk 3 above)
Comment (immeraufdemhund) [reply keyboards/handwired/dactyl_promicro/keymaps/default/keymap.c]: fixed

**Comment by immeraufdemhund** [comment]
any updates on this? @zvecr or @noroadsleft

**Comment by immeraufdemhund** [comment]
@zvecr @noroadsleft or @kuchosauronad0, I've reverted the breaking changes and would really like this merged in to benefit our website and sales so people can get a hex file from qmk_configurator.

**Comment by noroadsleft** [comment]
:tada: 

Thanks!

**Pull request merged**